misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt

```diff
@@ -44,80 +46,82 @@ class HibernateModule(
   val config = config.withDefaults()
 
   override fun configure() {
-    val configKey = DataSourceConfig::class.toKey(qualifier)
+    val sessionFactoryProvider = getProvider(keyOf<SessionFactory>(qualifier))
+    val environmentProvider: Provider<Environment> = getProvider(keyOf<Environment>())
 
-    val entitiesKey = setOfType(HibernateEntity::class).toKey(qualifier)
-    val entitiesProvider = getProvider(entitiesKey)
-
-    val environmentKey = Environment::class.toKey()
-    val environmentProvider = getProvider(environmentKey)
+    newMultibinder<DataSourceDecorator>(qualifier)
 
-    val eventListenersKey = setOfType(ListenerRegistration::class).toKey(qualifier)
-    val eventListenersProvider = getProvider(eventListenersKey)
+    bind<Query.Factory>().to<ReflectionQuery.Factory>()
+    bind<QueryLimitsConfig>()
+        .toInstance(QueryLimitsConfig(MAX_MAX_ROWS, ROW_COUNT_ERROR_LIMIT, ROW_COUNT_WARNING_LIMIT))
+    bind(keyOf<DataSourceConfig>(qualifier)).toInstance(config)
 
-    val sessionFactoryKey = SessionFactory::class.toKey(qualifier)
-    val sessionFactoryProvider = getProvider(sessionFactoryKey)
+    // Bind StartVitessService.
+    install(ServiceModule<StartVitessService>(qualifier))
+    bind(keyOf<StartVitessService>(qualifier)).toProvider(object : Provider<StartVitessService> {
+      @Inject lateinit var environment: Environment
+      override fun get(): StartVitessService {
+        return StartVitessService(environment = environment, config = config, qualifier = qualifier)
+      }
+    }).asSingleton()
 
-    val sessionFactoryServiceKey = SessionFactoryService::class.toKey(qualifier)
-    val sessionFactoryServiceProvider = getProvider(sessionFactoryServiceKey)
+    // Bind PingDataBaseService.
+    bind(keyOf<PingDatabaseService>(qualifier)).toProvider(Provider<PingDatabaseService> {
+      PingDatabaseService(config, environmentProvider.get())
+    }).asSingleton()
+    // TODO(rhall): depending on Vitess is a hack to simulate Vitess has already been started in the
+    // env. This is to remove flakiness in tests that are not waiting until Vitess is ready.
+    // This should be replaced with an ExternalDependency that manages vitess.
```

Comment:
This comment was _moved_ from `PingDatabaseService`. It looked important to preserve.